src/humanloop/client.py

```diff
@@ -78,11 +90,124 @@ def __init__(
+                attributes={
+                    "instrumentor": "humanloop.sdk",
+                }
+            ),
+        )
+        instrument_provider(provider=self._tracer_provider)
+        self._tracer_provider.add_span_processor(
+            HumanloopSpanProcessor(
+                exporter=HumanloopSpanExporter(
+                    client=self,
+                )
+            ),
+        )
+        tracer = self._tracer_provider.get_tracer("humanloop.sdk")
+        set_tracer(tracer)
+
         eval_client = ExtendedEvalsClient(client_wrapper=self._client_wrapper)
         eval_client.client = self
         self.evaluations = eval_client
         self.prompts = ExtendedPromptsClient(client_wrapper=self._client_wrapper)
 
+    def prompt(
+        self,
+        # TODO: Template can be a list of objects
+        path: Optional[str] = None,
+        model: Optional[str] = None,
+        endpoint: Optional[Literal["chat", "edit", "complete"]] = None,
+        template: Optional[str] = None,
+        provider: Optional[
+            Literal["openai", "openai_azure", "mock", "anthropic", "bedrock", "cohere", "replicate", "google", "groq"]
+        ] = None,
+        max_tokens: Optional[int] = None,
+        stop: Optional[Union[str, list[str]]] = None,
+        temperature: Optional[float] = None,
+        top_p: Optional[float] = None,
+        presence_penalty: Optional[float] = None,
+        frequency_penalty: Optional[float] = None,
+    ):
+        """Decorator to mark a function as a Humanloop Prompt.
+
+        The decorator intercepts calls to LLM provider APIs and uses them
+        in tandem with the template provided by the user to create a Prompt
+        in Humanloop.
+
+        Arguments:
+            path: Optional. The path where the Prompt is created. If not
+                provided, the function name is used as the path and
+                the File is created in the root of your Humanloop's
+                organization workspace.
+            template: The template for the Prompt. This is the text of
+                the system message used to set the LLM prompt. The template
+                accepts template slots using the format `{slot_name}`.
+
+                The text of the system message is matched against the template
+                to extract the slot values. The extracted values will be
+                available in the Log's inputs
+        """
+        return prompt_decorator(
+            path=path,
+            model=model,
+            endpoint=endpoint,
+            template=template,
+            provider=provider,
+            max_tokens=max_tokens,
+            stop=stop,
+            temperature=temperature,
+            top_p=top_p,
+            presence_penalty=presence_penalty,
+            frequency_penalty=frequency_penalty,
+        )
+
+    def tool(
+        self,
+        path: Optional[str] = None,
+        attributes: Optional[dict[str, typing.Any]] = None,
```

Comment:
Are we missing args that are supported by tool kernel?

Reply:
added setup_values

Reply:
Ack I do have a question: how is this supposed to work with the prompt? Should the user provide a dict of setup_values, which then can be referred in the tool e.g. an API key?

Reply:
> setup_values

I actually think setup_values are not relevant here because these are not tools we can run. So this should be removed

Reply:
setup_values was removed